### PR TITLE
Add Segment.RecipientCount

### PR DESCRIPTION
**Pull request by farkastamas84**
Hi,
I had some issues with querying segment recipients. "recipient_count" was not available, so total pages could not be calculated (API throws when querying non-existent page). So added Segment.RecipientCount.
Paging paramaters are optional according to the v3 spec., so changed recordsPerPage and page parameters to nullable and propose null for default values.

### CODE REVIEW

File: Source/StrongGrid/Model/Segment.cs [hunk 1]
```diff
@@ -42,5 +42,14 @@ public class Segment
...
 		[JsonProperty("conditions", NullValueHandling = NullValueHandling.Ignore)]
 		public SearchCondition[] Conditions { get; set; }
+
+		/// <summary>
+		/// Gets or sets the recipient count.
+		/// </summary>
+		/// <value>
+		/// The recipient count.
+		/// </value>
+		[JsonProperty("recipient_count", NullValueHandling = NullValueHandling.Ignore)]
```
Comment (Jericho) [inline Source/StrongGrid/Model/Segment.cs]: Looks like an oversight on my part. Thanks for adding this property.

**Comment by farkastamas84** [comment]
I think I was wrong about optional parameters, it seems they use internal default values.

Now I tried **Lists**.GetRecipientsAsync with nullable paging parameters, but I'm not getting all contacts. When sending null in recordsPerPage, but specifying 2 for page, I get the remaining ones.
In "Contacts API - Lists" specification, paging parameters are documented as optional, but it also mentions that page must be positive and page_size must be between 1 and 1000. So it seems paging itself is not optional. When not posting these params, they send the first page with some default page size.

I cannot test this with segments, since we have less than 1000 contacts per segment (Segment API does not mention a range though).

BTW, paging is quite strange. When querying Segment recipients with a page size of 100, I'm getting like 81, 114, 103, 95... contacts on consecutive pages. So not exactly the specified page size, but a bit more or a bit less.

**Comment by Jericho** [comment]
I agree paging is quite strange. And also it's not standard: everywhere else in the SendGrid API they accept `limit` and `offset`. It's quite strange that they don't use a different paging pattern for `GetRecipientsAsync`.

**Comment by Jericho** [comment]
For the time being, I prefer to stick with the non-nullable with parameters. The default values (100 for page size and 1 for page number) correspond to the default values mentioned in the documentation, therefore I think these default values are quite reasonable.

File: Source/StrongGrid/Resources/Segments.cs [hunk 2]
```diff
@@ -142,7 +142,7 @@ public Task DeleteAsync(long segmentId, bool deleteMatchingContacts = false, Can
 		/// <returns>
 		/// An array of <see cref="Contact" />.
 		/// </returns>
-		public Task<Contact[]> GetRecipientsAsync(long segmentId, int recordsPerPage = 100, int page = 1, CancellationToken cancellationToken = default(CancellationToken))
+		public Task<Contact[]> GetRecipientsAsync(long segmentId, int? recordsPerPage = null, int? page = null, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (Jericho) [inline Source/StrongGrid/Resources/Segments.cs]: Please revert to non-nullable parameters

Comment (farkastamas84) [inline Source/StrongGrid/Resources/Segments.cs]: Hi, reverted 20881152c4573d8773503b33d3705c76d9113e1c and renamed the PR.

**Comment by Jericho** [comment]
Perfect. Thanks

**Pull request merged**